CMakeLists.txt

```diff
@@ -615,10 +631,9 @@ set(MASTER_HEADERS
   ${PROJECT_SOURCE_DIR}/mfem.hpp
   ${PROJECT_SOURCE_DIR}/mfem-performance.hpp)
 
-set(_lib_path "${CMAKE_INSTALL_PREFIX}/lib")
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON CACHE BOOL "")
-set(CMAKE_INSTALL_RPATH "${_lib_path}" CACHE PATH "")
-set(CMAKE_INSTALL_NAME_DIR "${_lib_path}" CACHE PATH "")
+set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_FULL_LIBDIR}" CACHE PATH "")
+set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_FULL_LIBDIR}" CACHE PATH "")
```

Comment:
This should use `INSTALL_LIB_DIR` and not assume `GNUInstallDirs` is always used. This will require the defaults for `INSTALL_LIB_DIR` and similar to be moved up. If we want to have an option to use `GNUInstallDirs` then `INSTALL_LIB_DIR` can be defined based on what `GNUInstallDirs` defines.